tests/src/astro/electromagnetism/unitTestYarkovskyAcceleration.cpp

```diff
@@ -0,0 +1,207 @@
+
+    // Initial Conditions
+    const double initialSemiMajorAxis = 0.75 * AU;
+    const double initialEccentricity = 0.0;
+    const double initialInclination = 0.0;
+    const double initialRAAN = 0.0;
+    const double initialArgOfPeri = 0.0;
+    const double initialTrueAnomaly = 0.0;
+
+    // Bodies
+    std::vector< std::string > bodiesToCreate { "Sun" };
+    BodyListSettings bodySettings = getDefaultBodySettings( bodiesToCreate, frameOrigin, frameOrientation );
+    SystemOfBodies bodies = createSystemOfBodies( bodySettings );
+
+    // Asteroid
+    bodies.createEmptyBody( "Asteroid" );
+    bodies.at( "Asteroid" )->setConstantBodyMass( 0.0 );
+
+    // Define propagator settings variables.
+    std::vector< std::string > bodiesToPropagate { "Asteroid" };
+    std::vector< std::string > centralBodies { "Sun" };
+
+    // Accelerations
+    std::map< std::string, std::vector< std::shared_ptr< AccelerationSettings > > > accelerationsOfAsteroid = {
+            { "Sun", { pointMassGravityAcceleration( ), yarkovskyAcceleration( yarkovskyParameter ) }},
+    };
+
+    const double sunGravitationalParameter = bodies.at( "Sun" )->getGravitationalParameter( );
+
+    // Make the acceleration models
+    SelectedAccelerationMap accelerationSettings {{ "Asteroid", accelerationsOfAsteroid }};
+    basic_astrodynamics::AccelerationMap accelerationModels = createAccelerationModelsMap( bodies,
+                                                                                           accelerationSettings,
+                                                                                           bodiesToPropagate,
+                                                                                           centralBodies );
+
+    // Initial State
+    Eigen::VectorXd initialState = orbital_element_conversions::convertKeplerianToCartesianElements(
+            initialSemiMajorAxis,
+            initialEccentricity,
+            initialInclination,
+            initialRAAN,
+            initialArgOfPeri,
+            initialTrueAnomaly,
+            sunGravitationalParameter );
+
+    // Termination Setting
+    std::shared_ptr< propagators::PropagationTerminationSettings > terminationSettings = propagators::propagationTimeTerminationSettings(
+            simulationEndEpoch,
+            true );
+
+    // Integrator Settings
+    std::shared_ptr< numerical_integrators::IntegratorSettings< double>> integratorSettings = numerical_integrators::rungeKutta4Settings(
+            fixedStepSize );
+
+    // Propagator Settings
+    std::shared_ptr< propagators::TranslationalStatePropagatorSettings< double, double>> propagatorSettings = propagators::translationalStatePropagatorSettings(
+            centralBodies,
+            accelerationModels,
+            bodiesToPropagate,
+            initialState,
+            simulationStartEpoch,
+            integratorSettings,
+            terminationSettings,
+            propagators::cowell );
+
```

Comment:
I'd recommend one additional (lower-level) check in this test: save the yarkovsky acceleration as dependent variable, and check that it's:
* Colinear with the velocity vector (e.g. **a**\cdot**v**=||a||*||v||)  and that it changes proportionally with distance (take the value of ||r|| at t0, and then check the value of ||r|| at each later epoch, compute how the value of ||a|| should scale, and compare against the dependent variable result